lib/widgets/article/article-bookmark-feed.dart

```diff
@@ -0,0 +1,212 @@
+      articleModel.add(BookmarkedArticle.fromMap(mapList[i]));
+    }
+    return articleModel;
+  }
+
+  Future<void> hasBookMarkedArticles() async {
+    final db = await openDbConnection();
+
+    List<Map> isInDatabase = await db.rawQuery('SELECT * FROM bookmarks');
+
+    if (isInDatabase.length > 0) {
+      setState(() {
+        userHasBookMarkedArticles = true;
+      });
+    }
+  }
+
+  updateListView() async {
+    _articles = await getModelsFromMapList();
+
+    setState(() {
+      _articles = _articles;
+      count = _articles.length;
+    });
+  }
+
+  ListView populateListViewModel() {
+    return ListView.builder(
+        itemCount: count,
+        itemBuilder: (context, index) {
+          return ConstrainedBox(
+              constraints: BoxConstraints(minHeight: 150),
+              child: Container(
+                  decoration: BoxDecoration(
+                      border: Border(
+                          bottom: BorderSide(width: 2, color: Colors.white))),
+                  child: InkWell(
+                    onTap: () {
+                      Navigator.push(
+                          context,
+                          MaterialPageRoute(
+                              builder: (context) => ArticleBookmarkView(
+                                  article: this._articles[index])));
+                    },
+                    child: Column(
+                      children: [
+                        Row(
+                          children: [
+                            Expanded(
+                              flex: 8,
+                              child: Padding(
+                                padding: const EdgeInsets.all(25.0),
+                                child: Text(
+                                  truncateStr(
+                                      this._articles[index].articleTitle),
+                                  style: TextStyle(
+                                      color: Colors.white, fontSize: 18),
+                                ),
+                              ),
+                            ),
+                            Expanded(
+                                flex: 2,
+                                child: IconButton(
+                                    onPressed: () {
+                                      Navigator.push(
+                                          context,
+                                          MaterialPageRoute(
+                                              builder: (context) =>
+                                                  ArticleBookmarkView(
+                                                      article: this
+                                                          ._articles[index])));
+                                    },
+                                    icon: Icon(
+                                      Icons.arrow_forward_ios,
+                                      color: Colors.white,
+                                    )))
```

Comment:
Since the parent Inkwell has the onTap listener for the item, the IconButton can be replaced with just an Icon as there's no need of the children to have listeners for clicks/presses.

```suggestion
                                child: Icon(
                                      Icons.arrow_forward_ios,
                                      color: Colors.white,
                                    ))
```